js/classes/entities/midground/weapon/missile.js

```diff
@@ -0,0 +1,172 @@
+    }
+
+    static get DAMAGE() {
+        return 12;
+    }
+
+    /** Called when the missile collides with a block. */
+    onBlockCollision(block) {
+        if (!this.hasHit) {
+            this.hasHit = true;
+            this.explode();        
+        }
+    }
+
+    /** Called when the missile collides with an enemy. */
+    onEnemyCollision(enemy) {
+        // do nothing
+    }
+
+    /** Called when the missile collides with the player. */
+    onPlayerCollision(player) {
+        if (!this.hasHit) {
+            this.hasHit = true;
+            this.explode();
+        }
+    }
+
+    explode() {
+        ASSET_MGR.playSFX(SFX.EXPLOSION_BIG.path, SFX.EXPLOSION_BIG.volume);
+        GAME.addEntity(new ParticleEffect(this.pos, ParticleEffect.MISSILE_EXPLOSION));
+
+        // deal damage to nearby enemies
+        // NOTE: Yes, we could just expand the bomb's bounding box. I realized that after I implemented the getNearbyEntities function.
+        const nearbyEntities = getNearbyEntities(this.pos, 100);
+        for (const entity of nearbyEntities) {
+            if (entity == CHAD) { // ensure entity has a takeDamage method
+                entity.takeDamage(Missile.DAMAGE);
+            }
+        }
+
+        this.removeFromWorld = true;
+    }
+
+    loadAnimations() {
+        this.animations["firing"] = [];
+
+        this.animations["firing"]["right"] = new Animator(
+            Missile.SPRITESHEET, 
+            new Vector(0, Missile.SIZE.y * 0), 
+            Missile.SIZE, 2, 0.5); // 0 degree angle
+
+        this.animations["firing"]["rightDown"] = new Animator(
+            Missile.SPRITESHEET, 
+            new Vector(0, Missile.SIZE.y * 1), 
+            Missile.SIZE, 2, 0.5); // 45 degree angle
+
+        this.animations["firing"]["down"] = new Animator(
+            Missile.SPRITESHEET, 
+            new Vector(0, Missile.SIZE.y * 2), 
+            Missile.SIZE, 2, 0.5); // 90 degree angle
+
+        this.animations["firing"]["leftDown"] = new Animator(
+            Missile.SPRITESHEET, 
+            new Vector(0, Missile.SIZE.y * 3), 
+            Missile.SIZE, 2, 0.5); // 135 degree angle
+
+        this.animations["firing"]["left"] = new Animator(
+            Missile.SPRITESHEET, 
+            new Vector(0, Missile.SIZE.y * 4), 
+            Missile.SIZE, 2, 0.5); // 180 degree angle
+
+        this.animations["firing"]["leftUp"] = new Animator(
+            Missile.SPRITESHEET, 
+            new Vector(0, Missile.SIZE.y * 5), 
+            Missile.SIZE, 2, 0.5); // 225 degree angle
+
+        this.animations["firing"]["up"] = new Animator(
+            Missile.SPRITESHEET,
+            new Vector(0, Missile.SIZE.y * 6), 
+            Missile.SIZE, 2, 0.5); // 270 degree angle
+
+        this.animations["firing"]["rightUp"] = new Animator(
+            Missile.SPRITESHEET, 
+            new Vector(0, Missile.SIZE.y * 7), 
+            Missile.SIZE, 2, 0.5); // 315 degree angle
+    }
+
+    update() {
```

Comment:
Not a huge deal but don't we usually put the update method above the `loadAnimations()`? Might be good to stay consistent if that is correct.